src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -0,0 +1,131 @@
+            // Get the appSettings section.
+            var appSettings = (AppSettingsSection)config.GetSection("appSettings");
+
+            serverUrl = appSettings.Settings["ServerUrl"].Value;
+            if (String.IsNullOrWhiteSpace(serverUrl))
+                throw new ArgumentException();
+
+            port = appSettings.Settings["Port"].Value;
+            if (String.IsNullOrWhiteSpace(port))
+                throw new ArgumentException();
+
+            page = appSettings.Settings["Page"].Value;
+            if (String.IsNullOrWhiteSpace(page))
+                throw new ArgumentException();
+
+            authenticationProvider = dynamoAuthenticationProvider;
+            customNodeManager = dynamoCustomNodeManager;
+        }
+
+        #endregion
+
+        internal void Authenticate()
+        {
+            // Manager must be initialized in constructor.
+            if (authenticationProvider == null)
+                throw new Exception(Resource.AuthenticationErrorMessage);
+
+            authenticationProvider.Login();
+        }
+
+        /// <summary>
+        /// Sends workspace and its' dependencies to Flood.
+        /// </summary>
+        public void Send(IEnumerable<IWorkspaceModel> workspaces)
+        {
+            if (String.IsNullOrWhiteSpace(serverUrl) || String.IsNullOrWhiteSpace(authenticationProvider.Username))
+                throw new Exception(Resource.ServerErrorMessage);
+
+            if (authenticationProvider == null)
+                throw new Exception(Resource.AuthenticationErrorMessage);
+
+            string fullServerAdress = serverUrl + ":" + port;
+
+            if (reachClient == null)
+                reachClient = new WorkspaceStorageClient(authenticationProvider, fullServerAdress);
+
+            HomeWorkspace = workspaces.OfType<HomeWorkspaceModel>().First();
+            var functionNodes = HomeWorkspace.Nodes.OfType<Function>();
+
+            List<CustomNodeDefinition> dependencies = new List<CustomNodeDefinition>();
+            foreach (var node in functionNodes)
+            {
+                dependencies.AddRange(node.Definition.Dependencies);
+            }
+
+            CustomNodesWorkspaces = new List<CustomNodeWorkspaceModel>();
+            foreach (var dependency in dependencies)
+            {
+                CustomNodeWorkspaceModel customNodeWs;
+                var isWorkspaceCreated = customNodeManager.TryGetFunctionWorkspace(dependency.FunctionId, false, out customNodeWs);
+                if (isWorkspaceCreated && !CustomNodesWorkspaces.Contains(customNodeWs))
+                    CustomNodesWorkspaces.Add(customNodeWs);
+            }
+
+            var result = reachClient.Send(HomeWorkspace, CustomNodesWorkspaces);
+        }
+
+        /// <summary>
+        /// Used for tests. Do not set it directly.
+        /// </summary>
+        public void SetClient(IWorkspaceStorageClient client)
```

Comment:
Instead of this, let's add an overloaded constructor for `PublishModel` that passes in the the `WorkspaceStorageClient`.